src/vs/workbench/contrib/positronHoloViews/test/browser/positronHoloViewsService.test.ts

```diff
@@ -0,0 +1,133 @@
+import { IRuntimeSessionService } from 'vs/workbench/services/runtimeSession/common/runtimeSessionService';
+import { TestLanguageRuntimeSession } from 'vs/workbench/services/runtimeSession/test/common/testLanguageRuntimeSession';
+import { TestRuntimeSessionService } from 'vs/workbench/services/runtimeSession/test/common/testRuntimeSessionService';
+import { workbenchInstantiationService } from 'vs/workbench/test/browser/workbenchTestServices';
+
+class TestNotebookService implements Partial<INotebookService> {
+	getRenderers(): INotebookRendererInfo[] {
+		return [];
+	}
+
+	getPreferredRenderer(_mimeType: string): NotebookOutputRendererInfo | undefined {
+		// Doesn't matter what the renderer is, just that it exists.
+		return <NotebookOutputRendererInfo>{
+			id: 'positron-ipywidgets',
+			extensionId: new ExtensionIdentifier('vscode.positron-ipywidgets'),
+		};
+	}
+
+	*getStaticPreloads(_viewType: string): Iterable<INotebookStaticPreloadInfo> {
+		// Yield nothing.
+	}
+}
+
+suite('Positron - PositronHoloViewsService', () => {
+	const disposables = ensureNoDisposablesAreLeakedInTestSuite();
+
+	let positronHoloViewsService: PositronHoloViewsService;
+	let runtimeSessionService: TestRuntimeSessionService;
+	let notebookEditorService: INotebookEditorService;
+
+	setup(() => {
+		const instantiationService = workbenchInstantiationService(undefined, disposables);
+		instantiationService.stub(INotebookRendererMessagingService, disposables.add(instantiationService.createInstance(NotebookRendererMessagingService)));
+		notebookEditorService = disposables.add(instantiationService.createInstance(NotebookEditorWidgetService));
+		instantiationService.stub(INotebookEditorService, notebookEditorService);
+		instantiationService.stub(INotebookService, new TestNotebookService());
+		instantiationService.stub(IWebviewService, disposables.add(new WebviewService(instantiationService)));
+		instantiationService.stub(IPositronNotebookOutputWebviewService, instantiationService.createInstance(PositronNotebookOutputWebviewService));
+		runtimeSessionService = disposables.add(new TestRuntimeSessionService());
+		instantiationService.stub(IRuntimeSessionService, runtimeSessionService);
+		positronHoloViewsService = disposables.add(instantiationService.createInstance(PositronHoloViewsService));
+	});
+
+	async function createConsoleSession() {
+
+		// Start a console session.
+		const session = disposables.add(new TestLanguageRuntimeSession(LanguageRuntimeSessionMode.Console));
+		runtimeSessionService.startSession(session);
+
+		await timeout(0);
+
+		const out: {
+			session: TestLanguageRuntimeSession;
+			plotClient: WebviewPlotClient | undefined;
+		} = {
+			session, plotClient: undefined,
+		};
+
+		disposables.add(positronHoloViewsService.onDidCreatePlot(client => {
+			out.plotClient = client;
+		}));
+
+		return out;
+	}
+
+	test('console session: dependency messages are absorbed without emitting plot', async () => {
```

Comment:
IIUC the test also checks that the display message emits a plot. It's kinda mimicking what the kernel would do when showing a plot, maybe the name could reflect that?